### PR TITLE
store/tikv: Remove gcTaskWorker

### PR DESCRIPTION
**Pull request by MyonKeminta**
Signed-off-by: MyonKeminta <MyonKeminta@users.noreply.github.com>

### What problem does this PR solve? 

The `RangeTaskRunner` is implemented just like how `gcTaskWorker` did, to suit for all tasks that need to do operations to regions in a range. Now let's remove `gcTaskWorker` and replace it with `RangeTaskRunner`.

### What is changed and how it works?

This PR removed `gcTaskWorker` and uses `RangeTaskRunner` instead.

### Check List 

Tests 

 - Unit test
 - Integration test

### CODE REVIEW

**Comment by MyonKeminta** [comment]
The metrics is correct, but  since `RangeTaskRunner` doesn't calculates number of failed regions, so the last log saying the number of failed regions doesn't work. I think I need to update `RangeTaskRunner` to support this.

**Comment by MyonKeminta** [comment]
@jackysp This is already tested by `TestDoGC`.

**Comment by jackysp** [comment]
I found the coverage is a bit decreasing.

**Comment by MyonKeminta** [comment]
@jackysp I think that's caused by removal of code which is previously covered. PTAL again.
Also, I'll do other works to improve the coverage, but it's not this PR.

**Comment by MyonKeminta** [comment]
@disksing PTAL again. There's some more changes to let `RangeTaskRunner` able to calculate failed regions.

File: store/tikv/delete_range.go [hunk 1]
```diff
@@ -77,13 +77,13 @@ func (t *DeleteRangeTask) Execute(ctx context.Context) error {
 }
 
 // Execute performs the delete range operation.
-func (t *DeleteRangeTask) sendReqOnRange(ctx context.Context, r kv.KeyRange) (int, error) {
+func (t *DeleteRangeTask) sendReqOnRange(ctx context.Context, r kv.KeyRange) (RangeTaskStat, error) {
 	startKey, rangeEndKey := r.StartKey, r.EndKey
-	completedRegions := 0
+	var stat RangeTaskStat
```
Comment (AndreMouche) [inline store/tikv/delete_range.go]: I do not think to define a variable without init it is a good idea, is it a good habit in golang?

Comment (MyonKeminta) [reply store/tikv/delete_range.go]: It's field should be initialized as 0 by default here.

Comment (disksing) [reply store/tikv/delete_range.go]: Totally good habit in Go.

**Pull request merged**